### PR TITLE
Add backward_passes_per_step for gradient aggregation with partial distributed optimizer

### PR DESCRIPTION
**Pull request by MrAta**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [x] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description
Add backward_passes_per_step for gradient aggregation with partial distributed optimizer.

Fixes # (issue).
N/A

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/GOVERNANCE.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

**Comment by MrAta** [comment]
Hi @romerojosh @maxhgerlach 

While adding this I was thinking where  the `PartialDistributedOptimzier` should be placed as we have three `DistributedOptmizer` defined in `horovod/tensorflow/__init__.py`, `horovod/tensorflow/keras/__init__.py`, and `horovod/keras/__init__.py`? 

Given that it's just a wrapper around DistributedOptimizer, feels like it could be at all three places; but I don't quite remember why I added it in `tensorflow/keras/__init__.py` when sending #3738 .

Can you please provide your input here? especially on the difference between horovod/keras and horovod/tensorflow/keras.

As an example, looking  #2346 (and related commits) seems the `backward_passes_per_step` is never given to the DistributedOptimizer in horovod/keras one.

**Comment by maxhgerlach** [comment]
Sorry for the long wait, @MrAta!

Could you rebase your branch to master? That should fix the CI problems.

The proposed changes look good to me

> While adding this I was thinking where the `PartialDistributedOptimzier` should be placed as we have three `DistributedOptmizer` defined in `horovod/tensorflow/__init__.py`, `horovod/tensorflow/keras/__init__.py`, and `horovod/keras/__init__.py`?
> 
> Given that it's just a wrapper around DistributedOptimizer, feels like it could be at all three places; but I don't quite remember why I added it in `tensorflow/keras/__init__.py` when sending #3738 .
> 
> Can you please provide your input here? especially on the difference between horovod/keras and horovod/tensorflow/keras.
> 
> As an example, looking #2346 (and related commits) seems the `backward_passes_per_step` is never given to the DistributedOptimizer in horovod/keras one.

I don't have a strong opinion here, but I don't actively use much of Keras myself. Generally it seems that not every feature from `hvd.tensorflow` has been replicated in Keras-specific wrappers. I assume some redundancies could be removed from the code base nowadays if there is interest in systematizing it more.

**Comment by MrAta** [comment]
Thanks for the update @maxhgerlach just rebased.

Re the redundancies, I will reach out to you and @romerojosh on slack to discuss it more. Thanks!

**Review by maxhgerlach — APPROVED** [review]
Thank you
